VMEncryption/volume-notif-svc/UdevVolNotif.c

```diff
@@ -235,32 +237,41 @@ void printdq(std::deque<device>&dq){
 
 int main(int argc, char *argv[]) {
 
-    prepare_log_file();
-
-    custom_log("\n\n### Azure Disk Encryption volume notification service ###");
-    custom_log("Starting udev volume notification program (%s) with %d args", argv[0], argc);
-    signal(SIGINT, sigint_handler);
-    custom_log("Registered SIGINT handler");
-
     // Command line options: -d for daemon mode
+    //-l is for log path.
     int c = 0;
     int daemon_mode = 0;
-    while ((c = getopt(argc, argv, "d")) != -1) {
+    std::string log_path="";
+    bool invalid_option=false;
+    while ((c = getopt(argc, argv, "dl:")) != -1) {
         switch (c) {
             case 'd':
-                custom_log("Option for switching to daemon mode provided");
                 daemon_mode = 1;
                 break;
+            case 'l':
+                 log_path=optarg;
+                 break;
             default:
-                custom_log("Unknown option %c", c);
-                exit(1);
+                invalid_option=true;
+                break;
         }
     }
 
+    prepare_log_file(log_path);
+    if(invalid_option==true){
+        custom_log("Unknown option %c", c);
+        exit(1);
+    }
+    custom_log("\n\n### Azure Disk Encryption volume notification service ###");
+    custom_log("Starting udev volume notification program (%s) with %d args", argv[0], argc);
+    signal(SIGINT, sigint_handler);
+    custom_log("Registered SIGINT handler");
+
     char current_working_directory[1024];
     getcwd(current_working_directory,1024);
-    
+
     if (daemon_mode) {
+        custom_log("Option for switching to daemon mode provided");
         daemonize(argc, argv);
     }
 
```

Comment:
if the partition is formatted as ext4; but not added for /etc/fstab yet due to a race; ADE is started by VNS; however, the partition is skipped as there is no mount point.

I think we should have mnt_monitor() as well to cover that scenario.

Reply:
Take a look at this example: https://github.com/canfikret/linux/blob/main/mount/mnt_monitor.cpp